Add HTTP Bearer Auth

mgenereu opened the pull request:
Closes psf/requests#4437.

sigmavirus24 reviewed (changes requested):
Please add some tests

mgenereu commented:
> Please add some tests

I can't get dev requirements to install.  Says it cannot connect to git for betamax but that's where I'm cloning from.  Are the tests available as GitHub actions or equiv.?

mgenereu commented:
> You'll want to update `git+git://github.com/sigmavirus24/betamax` to `git+https://github.com/betamaxpy/betamax`

Worked, fixed the tests, and added this line in too.
